In [1]:
# load required libraries or packages for this example
library(mlpack)
library(data.table)
library(wordcloud)
library(tm)
library(dplyr)
library(ggplot2)


Attaching package: ‘mlpack’


The following object is masked from ‘package:stats’:

    kmeans


The following object is masked from ‘package:base’:

    det


Loading required package: RColorBrewer

Loading required package: NLP


Attaching package: ‘dplyr’


The following objects are masked from ‘package:data.table’:

    between, first, last


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



Attaching package: ‘ggplot2’


The following object is masked from ‘package:NLP’:

    annotate




In [3]:
# read movie and ratings data into memory 
ratings <- fread("http://www.mlpack.org/datasets/ml-20m/ratings-only.csv.gz")
movies <- fread("http://www.mlpack.org/datasets/ml-20m/movies.csv.gz")

Lets check the data set to see if there are any missing values.

In [4]:
checkNas <- function(df)
{
    df      %>%
    is.na() %>%
    any()
}

checkNas(ratings)
checkNas(movies)

[1] FALSE

[1] FALSE

In [5]:
# Peak at or print the first 10 rows of the ratings data. 
head(ratings, n = 10)

userId movieId rating
1  1        1     3.5   
2  1       28     3.5   
3  1       31     3.5   
4  1       46     3.5   
5  1       49     3.5   
6  1      110     3.5   
7  1      149     4.0   
8  1      220     4.0   
9  1      250     4.0   
10 1      257     4.0

User 1 has rated the movies with id, 1 , 28, 31, 46, 49, 110 with a rating of 3.5. On the other hand the same user rated the movies with id, 149, 220, 250, 257 with a rating of 4. 

In [6]:
# Peak at or print the first 10 rows of the movie data.
head(movies, n = 10)

V1 movieId title                             
1  0  0       Toy Story (1995)                  
2  1  1       Jumanji (1995)                    
3  2  2       Grumpier Old Men (1995)           
4  3  3       Waiting to Exhale (1995)          
5  4  4       Father of the Bride Part II (1995)
6  5  5       Heat (1995)                       
7  6  6       Sabrina (1995)                    
8  7  7       Tom and Huck (1995)               
9  8  8       Sudden Death (1995)               
10 9  9       GoldenEye (1995)                  
   genres                                     
1  Adventure|Animation|Children|Comedy|Fantasy
2  Adventure|Children|Fantasy                 
3  Comedy|Romance                             
4  Comedy|Drama|Romance                       
5  Comedy                                     
6  Action|Crime|Thriller                      
7  Comedy|Romance                             
8  Adventure|Children                         
9  Action                                     
10 Action|Adventure|Thriller

The movielens dataset contains data on a large number of movies. Present are three columns: movie ID, title, and genres. The genres fall into 18 categories, but more on that later.

In [14]:
# create a corpus of the titles and clean up the titles 
options(warn=-1)
cleanedCorpus <- function(cp)
{
    
    cp %>% 
    VectorSource() %>%
    Corpus()       %>%
    tm_map(. , content_transformer(tolower)) %>%
    tm_map(. , removeWords, stopwords(kind = 'english')) %>%
    tm_map(. , removePunctuation) %>%
    tm_map(. , removeNumbers)
}

movie_corpus <- cleanedCorpus(movies$title)

In [15]:
inspect(movie_corpus)


IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)

